tests/test_handler_mlflow.py

```diff
@@ -230,6 +235,52 @@ def test_multi_thread(self):
                 self.tmpdir_list.append(res)
                 self.assertTrue(len(glob.glob(res)) > 0)
 
+    @skip_if_quick
+    @unittest.skipUnless(has_dataset_tracking, reason="Requires mlflow version >= 2.4.0.")
+    def test_dataset_tracking(self):
+        test_bundle_name = "endoscopic_tool_segmentation"
+        with tempfile.TemporaryDirectory() as tempdir:
+            resource = "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/endoscopic_tool_dataset.zip"
+            md5 = "f82da47259c0a617202fb54624798a55"
+            compressed_file = os.path.join(tempdir, "endoscopic_tool_segmentation.zip")
+            data_dir = os.path.join(tempdir, "endoscopic_tool_dataset")
+            if not os.path.exists(data_dir):
+                download_and_extract(resource, compressed_file, tempdir, md5)
+
+            download(test_bundle_name, bundle_dir=tempdir)
+
+            bundle_root = os.path.join(tempdir, test_bundle_name)
+            config_file = os.path.join(bundle_root, "configs/inference.json")
+            meta_file = os.path.join(bundle_root, "configs/metadata.json")
+            logging_file = os.path.join(bundle_root, "configs/logging.conf")
+            workflow = ConfigWorkflow(
+                workflow="infer",
+                config_file=config_file,
+                meta_file=meta_file,
+                logging_file=logging_file,
+                init_id="initialize",
+                run_id="run",
+                final_id="finalize",
+            )
+            tracking_path = os.path.join(bundle_root, "eval")
+            mlflow_handler = MLFlowHandler(
+                iteration_log=False, epoch_log=False, dataset_log=True, tracking_uri=path_to_uri(tracking_path)
+            )
+
+            workflow.bundle_root = bundle_root
+            workflow.dataset_dir = data_dir
+            workflow.initialize()
+            mlflow_handler.attach(workflow.evaluator)
+            workflow.run()
+            workflow.finalize()
+
+            cur_run = mlflow_handler.client.get_run(mlflow_handler.cur_run.info.run_id)
+            logged_nontrain_set = [
+                x for x in cur_run.inputs.dataset_inputs if "nontrain" == x.dataset.name[: len("nontrain")]
```

Comment:
And I didn't get where we set the name `nontrain` for `x.dataset`?

Thanks.

Reply:
Updated this part.